vignettes/workflows.Rmd

```diff
@@ -0,0 +1,210 @@
+
+What we want is that after a particular version of a report to have been run, it is easy for other people to start depending on it.  To achieve this we need a method for getting the archive from one person's computer to another.
+
+For this discussion we will assume two researchers ["Alice" and "Bob"](https://en.wikipedia.org/wiki/Alice_and_Bob) rather than "Researcher A" and "Researcher B".
+
+Alice develops a report `report-a` on her computer by running `orderly::orderly_new("report-a")` and working with `orderly::orderly_develop_start()` and `orderly::orderly_run()` until she is happy with the results.  She then creates a pull request, and Bob approves it (probably after checking that the report runs on his machine too) and merged it.
+
+Now that this report has been merged into master we need to run a "canonical copy".  We do this on a third copy of the sources that are checked out somewhere accessible to *both* Alice and Bob; this might be using Dropbox or another file-sharing platform, or on a network drive visible to both users.
+
+Either Alice or Bob may now pull the new changes onto this new source copy, and run `orderly run report-a` on the command line.  Using the command-line version of `orderly` ensures that the report is run in a clean session, along with keeping a copy of the log alongside the outputs.
+
+Now we have both Alice and Bob able to get the *source* of `report-a` and our shared archive having a copy of it; how do we get it to Alice and Bob?  We need to edit the `orderly_config.yml` to tell it about this archive:
+
+```yaml
+remote:
+  default:
+    driver: orderly::orderly_remote_path
+    args:
+      path: $ORDERLY_PATH_DEFAULT_REMOTE
+```
+
+This tells `orderly` that we will use a file-based remote source (i.e., a shared copy of orderly) and that the path to this source is determined by the [Environment variable](https://en.wikipedia.org/wiki/Environment_variable) `ORDERLY_PATH_DEFAULT_REMOTE`.  Using an environment variable is necessary here as this path will certainly differ between Alice and Bob.
+
+To fill in the environment variable we create a file `orderly_envir.yml` (which should be added to `.gitignore`) containing the path to the shared source.  For Alice, this might look like
+
+```yaml
+ORDERLY_PATH_DEFAULT_REMOTE: /Users/alice/Dropbox/ourorderly
+```
+
+and for Bob it might look like
+
+```yaml
+ORDERLY_PATH_DEFAULT_REMOTE: c:/Bob/Documents/Dropbox/ourorderly
+```
+
+With this done, Bob can get a copy of `report-a` by running
+
+```r
+orderly::orderly_pull_archive("report-a")
+```
+
+which will copy the latest version of the report from the shared archive into their personal archive.  A this point, Bob has *exactly* the copy of `report-a` that was run on the shared archive.  In the same way that Alice and Bob can exchange git sha hashes to ensure that they are on the same version of the source tree, they can now exchange report ids and ensure that they are looking at the same output.
```

Comment:
What's your favourite way of showing the hash? `git rev-parse HEAD`?

Reply:
yes, it's madness that this is not easier to do